metricbeat/module/jolokia/jmx/jmx_integration_test.go

```diff
@@ -33,25 +33,24 @@ func TestFetch(t *testing.T) {
 	compose.EnsureUp(t, "jolokia")
 
 	for _, config := range getConfigs() {
-		f := mbtest.NewEventsFetcher(t, config)
-		events, err := f.Fetch()
-		if !assert.NoError(t, err) {
-			t.FailNow()
-		}
+		f := mbtest.NewReportingMetricSetV2Error(t, config)
+		events, errs := mbtest.ReportingFetchV2Error(f)
+		assert.Empty(t, errs)
+		assert.NotEmpty(t, events)
 		t.Logf("%s/%s events: %+v", f.Module().Name(), f.Name(), events)
-		if len(events) == 0 || len(events[0]) <= 1 {
-			t.Fatal("Empty events")
-		}
 	}
 }
 
 func TestData(t *testing.T) {
 	compose.EnsureUp(t, "jolokia")
```

Comment:
I'm coming late with this comment, please forgive me :slightly_smiling_face: Following the reasoning here https://github.com/elastic/beats/pull/11114 I'll delete the `compose.EnsureUp` line and lines from 49 to 51 to avoid flakys in CI. It will also save some time if compose doesn't find the container and has to start it again to simply omit this test in line 53. 

Offtopic: I'm also not sure exactly what's the original idea (before this PR) in this module of having an slice of configs that was overwriting `data.json` on each call to `WriteEventsReporterV2`

Final idea is that it looks like this:

```go
func TestData(t *testing.T) {
    for _, config := range getConfigs() {
        ms := mbtest.NewReportingMetricSetV2(t, config)
        err := mbtest.WriteEventsReporterV2(ms, t, "")
        if err != nil {
            t.Fatal("write", err)
        }
    }
}
```

Reply:
Offtopic part: seems like having an slice of configs is to test different jmx mapping and especially with different `http_method`. Im not a fan of the overwriting `data.json` part because if I actually let it overwrite three times, in `data.json` the `uptime` field will disappear with the last config. In order to keep `uptime` in `data.json`, I had to manually only let TestData to write to data.json with the first config...

Reply:
Thanks for your explanation on why we should remove `compose.EnsureUp` here. So in the future, if users want to regenerate `data.json` without a local apache tomcat, we just need to run TestFetch first to start the container.